### PR TITLE
Fixes #20692 - Add API for rerunning jobs

### PR DESCRIPTION
**Pull request by adamruzicka**
- [x] tests

### CODE REVIEW

**Comment by iNecas** [comment]
Setting `Waiting on contributor` until the tests are added. Haven't tested yet, but looks good code-wise, except missing permissions definition (which is the reason why the tests are failing)

**Comment by iNecas** [comment]
👮  is not happy

**Comment by adamruzicka** [comment]
[test]

**Comment by iNecas** [comment]
We got conflicts, please rebase and run tests manually, as we have broken master at the moment

**Comment by iNecas** [comment]
Backtrace:

```
| Foreman::Exception: ERF42-4343 [Foreman::Exception]: Cannot resolve hosts without a user
 | /home/inecas/Projects/ws/foreman-rex/foreman_remote_execution/app/models/targeting.rb:35:in `resolve_hosts!'
 | /home/inecas/Projects/ws/foreman-rex/foreman_remote_execution/app/lib/actions/remote_execution/run_hosts_job.rb:22:in `plan'
 | /home/inecas/Projects/ws/foreman-rex/dynflow/lib/dynflow/action.rb:473:in `block (3 levels) in execute_plan'
```

**Comment by iNecas** [comment]
Adding link to hammer for better orientation https://github.com/theforeman/hammer_cli_foreman_remote_execution/pull/18

File: app/models/job_invocation_composer.rb [hunk 1]
```diff
@@ -194,10 +194,12 @@ def concurrency_control_params
     end
 
     def targeting_params
+      base = { :user_id => job_invocation.targeting.user_id }
```
Comment (adamruzicka) [inline app/models/job_invocation_composer.rb]: Maybe it would make sense to use `User.current` here instead of copying over the original user

Comment (iNecas) [reply app/models/job_invocation_composer.rb]: Definitely should be the current user

**Comment by iNecas** [comment]
Running tests manually http://ci.theforeman.org/job/test_plugin_pull_request/4760/

**Comment by iNecas** [comment]
Tested and works like a charm. The relevant tests in CI work as well. Thanks @adamruzicka

**Pull request merged**